nvidia_gpu_stats/cmd/nvidia_gpu_stats/main.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/process"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+type GPUNvidia struct {
+	pid      int
+	nvmlInit nvml.Return
+}
+
+func NewGPUNvidia(pid int) *GPUNvidia {
+	g := &GPUNvidia{pid: pid}
+	return g
+}
+
+func (g *GPUNvidia) gpuInUseByProcess(device nvml.Device) bool {
+	proc, err := process.NewProcess(int32(g.pid))
+	if err != nil {
+		return false
+	}
+
+	ourPids := make(map[int32]struct{})
+	ourPids[int32(g.pid)] = struct{}{}
+
+	childProcs, err := proc.Children()
+	if err == nil {
+		for _, childProc := range childProcs {
+			ourPids[childProc.Pid] = struct{}{}
+		}
+	}
+
+	computeProcesses, ret := device.GetComputeRunningProcesses()
+	if ret != nvml.SUCCESS {
+		return false
+	}
+	graphicsProcesses, ret := device.GetGraphicsRunningProcesses()
+	if ret != nvml.SUCCESS {
+		return false
+	}
+	pidsUsingDevice := make(map[int32]struct{})
+	for _, p := range computeProcesses {
+		pidsUsingDevice[int32(p.Pid)] = struct{}{}
+	}
+	for _, p := range graphicsProcesses {
+		pidsUsingDevice[int32(p.Pid)] = struct{}{}
+	}
+
+	intersectionCount := 0
+	for pid := range pidsUsingDevice {
+		if _, exists := ourPids[pid]; exists {
+			intersectionCount++
+		}
+	}
+
+	return intersectionCount > 0
+}
+
+func (g *GPUNvidia) SampleMetrics() map[string]any {
```

Comment:
doc: what's the format of the map that it returns? is it nested? are the values guaranteed to have specific types? are the keys arbitrary, or always the same?